In [12]:
import pandas as pd

rA= pd.read_csv('rAnew2.csv')
rA

날짜   월  사고수   날씨
0     1   1   53  1.0
1     2   1   95  NaN
2     3   1   79  NaN
3     4   1   78  NaN
4     5   1   68  NaN
..   ..  ..  ...  ...
361  27  12   40  2.0
362  28  12   81  2.0
363  29  12   71  1.0
364  30  12   89  NaN
365  31  12   98  1.0

[366 rows x 4 columns]

In [13]:
rA.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 366 entries, 0 to 365
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   날짜      366 non-null    int64  
 1   월       366 non-null    int64  
 2   사고수     366 non-null    int64  
 3   날씨      148 non-null    float64
dtypes: float64(1), int64(3)
memory usage: 11.6 KB


In [9]:
rAwinter = rA.drop(labels=[83,356], axis=0 ,inplace=True)
rAwinter

In [11]:
print(rAwinter)

None


In [20]:
rAwinter = rA.iloc[0:82]
rAwinter

날짜  월  사고수   날씨
0    1  1   53  1.0
1    2  1   95  NaN
2    3  1   79  NaN
3    4  1   78  NaN
4    5  1   68  NaN
..  .. ..  ...  ...
77  18  3   87  NaN
78  19  3   77  2.0
79  20  3  100  NaN
80  21  3   92  NaN
81  22  3   66  NaN

[82 rows x 4 columns]

In [17]:
rAwinter2 = rA.iloc[336:366]
rAwinter2

날짜   월  사고수   날씨
336   2  12   74  NaN
337   3  12   78  NaN
338   4  12  115  NaN
339   5  12   76  NaN
340   6  12   61  NaN
341   7  12   82  NaN
342   8  12  105  NaN
343   9  12   73  NaN
344  10  12   85  3.0
345  11  12  100  NaN
346  12  12   64  NaN
347  13  12   45  3.0
348  14  12   74  NaN
349  15  12   88  NaN
350  16  12   89  NaN
351  17  12  101  1.0
352  18  12  142  1.0
353  19  12  114  NaN
354  20  12   54  NaN
355  21  12  108  NaN
356  22  12  102  NaN
357  23  12  108  2.0
358  24  12   95  NaN
359  25  12   63  NaN
360  26  12   52  NaN
361  27  12   40  2.0
362  28  12   81  2.0
363  29  12   71  1.0
364  30  12   89  NaN
365  31  12   98  1.0

In [21]:
rAwinReal = pd.concat([rAwinter, rAwinter2], axis = 0)

In [22]:
rAwinReal

날짜   월  사고수   날씨
0     1   1   53  1.0
1     2   1   95  NaN
2     3   1   79  NaN
3     4   1   78  NaN
4     5   1   68  NaN
..   ..  ..  ...  ...
361  27  12   40  2.0
362  28  12   81  2.0
363  29  12   71  1.0
364  30  12   89  NaN
365  31  12   98  1.0

[112 rows x 4 columns]

In [31]:
rAwinReal.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 112 entries, 0 to 365
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   날짜      112 non-null    int64  
 1   월       112 non-null    int64  
 2   사고수     112 non-null    int64  
 3   날씨      39 non-null     float64
dtypes: float64(1), int64(3)
memory usage: 4.4 KB


In [32]:
rAwinReal['사고수'].astype(int)

0      53
1      95
2      79
3      78
4      68
       ..
361    40
362    81
363    71
364    89
365    98
Name: 사고수, Length: 112, dtype: int32

In [33]:
rAwinReal.sum()

날짜      1679.0
월        515.0
사고수    10036.0
날씨        74.0
dtype: float64

In [25]:
snow = rAwinReal['날씨'] == 1

a = rAwinReal[snow]


a

날짜   월  사고수   날씨
0     1   1   53  1.0
33    3   2  115  1.0
34    4   2  104  1.0
37    7   2  159  1.0
38    8   2  106  1.0
39    9   2   86  1.0
40   10   2  100  1.0
47   17   2  114  1.0
351  17  12  101  1.0
352  18  12  142  1.0
363  29  12   71  1.0
365  31  12   98  1.0

In [41]:
a.sum()

날짜      154.0
월        63.0
사고수    1249.0
날씨       12.0
dtype: float64

In [42]:
# 12일중 1249건. 확실히 눈이 올때는 사고가 많이 난다.

In [45]:
rain = rAwinReal['날씨'] == 2

b = rAwinReal[rain]

# b

In [39]:
b.sum()

날짜      372.0
월        69.0
사고수    1600.0
날씨       38.0
dtype: float64

In [40]:
# 19일 중 1600건

In [27]:
rsAll = rAwinReal['날씨'] == 3

c = rAwinReal[rsAll]

c

날짜   월  사고수   날씨
5     6   1  113  3.0
7     8   1   93  3.0
18   19   1   82  3.0
46   16   2   98  3.0
52   22   2   90  3.0
63    4   3   66  3.0
344  10  12   85  3.0
347  13  12   45  3.0

In [37]:
c.sum()

날짜      98.0
월       34.0
사고수    672.0
날씨      24.0
dtype: float64

In [38]:
# 8일중 사고 672건

In [43]:
rAw = rAwinReal.dropna()
# rAw

In [34]:
rAw.to_csv('rAwinter.csv')

In [29]:
rAw['사고수'].astype(int)

0       53
5      113
6      123
7       93
18      82
26      66
27      78
30      93
33     115
34     104
37     159
38     106
39      86
40     100
42     100
45      91
46      98
47     114
51     115
52      90
55      80
56      83
58      82
59      74
60      48
61      69
63      66
69      92
72     100
78      77
344     85
347     45
351    101
352    142
357    108
361     40
362     81
363     71
365     98
Name: 사고수, dtype: int32

In [44]:
# rAw

In [30]:
rAw.sum()

날짜      624.0
월       166.0
사고수    3521.0
날씨       74.0
dtype: float64

39일의 눈비, 눈, 비 오는 날이 있다. 사고수는 3521건.

전체 날짜는 112일 사고수는 10036

In [46]:
# 눈비 8일중 사고 672건
# 비 19일 중 1600건

# 눈 12일중 1249건. 확실히 눈이 올때는 사고가 많이 난다.

#### 겨울에도 눈이 오면 사고 발생확률이 높아진다.